crypto/asn1/x_long.c

```diff
@@ -56,11 +56,30 @@ static void long_free(ASN1_VALUE **pval, const ASN1_ITEM *it)
     *(long *)pval = it->size;
 }
 
+static int num_bits_ulong(unsigned long value)
```

Comment:
As for one being more efficient than the other. Having one operating on uint64_t would definitely *not* be most efficient on 32-bit platforms, quite contrary. Secondly with such small function one shouldn't underestimate effect of inlining and code proximity... Either way I'd argue that if it is to be solved it better be solved separately...

Fun fact in the context. Consider
```
void foo() {}
void bar() { foo(); }
```
Examine code generated without -fPIC and *with*...